python/ray/data/tests/test_file_based_datasource.py

```diff
@@ -118,6 +118,30 @@ def test_windows_path():
         assert _is_local_windows_path("c:\\some\\where/mixed")
 
 
+@pytest.mark.parametrize(
+    "shuffle, valid",
+    [
+        (None, True),
+        ("files", True),
+        (True, False),
+        (False, False),
+        ("file", False),
+    ],
+)
+def test_shuffle_arg(ray_start_regular_shared, tmp_path, shuffle, valid):
+
+    path = os.path.join(tmp_path, "test.txt")
+    with open(path, "w"):
+        pass
+
+    if valid:
+        FileBasedDatasource(path, shuffle=shuffle)
+
+    else:
+        with pytest.raises(ValueError):
+            FileBasedDatasource(path, shuffle=shuffle)
+
+
```

Comment:
Nit: IMO we should split this up into two tests like `test_invalid_shuffle_arg_raises_error` and `test_valid_shuffle_arg_does_not_raise_error`.

In general, descriptive test names and avoiding conditionals are good unit test practices.